Feature/refatory api

edilson14 opened the pull request:
refatoring api calls and export models from a single file to simplife some code

edilson14 committed:
refator(api): refatoring request

edilson14 committed:
refator: export models from single file

edilson14 committed:
refator: remove static route class

![effetive dart](https://dart.dev/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members)

edilson14 committed:
refator(routes): rename routes

edilson14 committed:
refator: improve funcition name

edilson14 committed:
fix: lyrics function called on class instance

edilson14 committed:
fix: genius class doesnt inherit http client

remove heritage to prevente acess to unecessary methods at genius

hbtalha commented on lib/src/genius.dart:
```diff
@@ -235,8 +255,8 @@ class Genius {
   /// Uses beautiful_soup to scrape song lyrics off of a Genius song URL
-  static Future<String?> lyrics({required String url}) async {
-    String responseBody = (await http.get(Uri.parse(Uri.encodeFull(url)))).body;
+  Future<String?> lyrics({required String url}) async {
```
This function should be static

edilson14 replied:
![image](https://github.com/hbtalha/genius_lyrics/assets/32166914/49b1732b-f5f7-430a-b52a-deb235571d54)
why? if we put it static we have to do requestBody static too , and there's no need to do that , because we already create an instance of httclient

edilson14 replied:
@hbtalha here

hbatalhaStch replied:
This function has to be static, should revert it so it doesn't use the instance of HttpClient

hbatalhaStch replied:
This function is public and it doesn't need an access token, that's why it should be staic

edilson14 replied:
![image](https://github.com/hbtalha/genius_lyrics/assets/32166914/c2971fd4-8ae8-44aa-bc2f-86622e33c074)
back to client , he's supost to take care of all api request , to make this method static we have to create our own parseUrl because we can't use non static methods inside statics 

this break the DRY ( Don't Repeat YourSelf ) principle , just to make this method static ?

hbtalha replied:
It is not always that you are able to follow the DRY strictly.
In this case, it is worth noting that the `requestBody` function as of right now is only used in this function.

I am aware of the implications of keeping this function static and it is stated in the previous comments the reasons why this function should remain static.

edilson14 replied:
i made a change and bonus, i follow the DRY :') , and the method will be static

hbtalha commented on lib/api/client.dart:
```diff
@@ -0,0 +1,40 @@
+    }
+  }
+
+  Future<String> requstBody({required String url}) async {
```
I believe there's a typo here (a missing letter `e`), shouldn't it be `requestBody`?

edilson14 replied:
you're right

edilson14 committed:
fix: function name
https://github.com/hbtalha/genius_lyrics/pull/16/files/ada6457c0c6b13da6e1dd3c8a85ae077573bac02#r1365458653

edilson14 committed:
test: fix line break expected

edilson14 committed:
feature: keep search lyrics static

edilson14 commented:
i made changes to keep search lyrics static

hbtalha commented:
Thank you for the PR 😀
